pkgs/applications/editors/android-studio/common.nix

```diff
@@ -110,7 +111,20 @@ let
         ]}" \
         --set QT_XKB_CONFIG_ROOT "${xkeyboard_config}/share/X11/xkb" \
         --set FONTCONFIG_FILE ${fontsConf}
+
+        install -Dm644 bin/studio.png $out/share/pixmaps/${drvName}.png
+        ln -s ${desktopItem}/share/applications $out/share/applications
     '';
+
+    desktopItem = makeDesktopItem rec {
+      name = drvName;
+      exec = pname;
+      icon = drvName;
+      desktopName = "Android Studio";
+      comment = "The official Android IDE";
+      categories = "Development;IDE;";
+    };
```

Comment:
Setting `startupNotify = true;` and `extraEntries="StartupWMClass=jetbrains-studio";` might be helpful as well.